### PR TITLE
more: add arguments print over and clean print

### PR DESCRIPTION
**Pull request by Ideflop**
#2320 

This pull request adds two arguments: `-c`, `--print-over` and `-p`, `--clean-print`. 

I have included them in an `if ... else if` block because when both `-c` and `-p` are provided, it results in the `--print-over` options being applied.

### CODE REVIEW

**Comment by sylvestre** [comment]
The test is failing:
```

--- TRY 3 STDERR:        coreutils::tests test_more::test_valid_arg ---
thread 'test_more::test_valid_arg' panicked at 'Command was expected to succeed.
stdout = 
 stderr = thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 6, kind: Uncategorized, message: "No such device or address" }', src/uu/more/src/more.rs:223:33
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
', tests/by-util/test_more.rs:15:27
stack backtrace:
   0: rust_begin_unwind
             at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/panicking.rs:142:14
   2: tests::common::util::CmdResult::success
             at ./tests/common/util.rs:390:9
   3: tests::common::util::UCommand::succeeds
             at ./tests/common/util.rs:1559:9
   4: tests::test_more::test_valid_arg
             at ./tests/by-util/test_more.rs:15:5
   5: tests::test_more::test_valid_arg::{{closure}}
             at ./tests/by-util/test_more.rs:14:1
   6: core::ops::function::FnOnce::call_once
             at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/ops/function.rs:248:5
   7: core::ops::function::FnOnce::call_once
             at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/ops/function.rs:248:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

File: tests/by-util/test_more.rs [hunk 1]
```diff
@@ -10,6 +10,18 @@ fn test_more_no_arg() {
     }
 }
 
+#[test]
+fn test_valid_arg() {
+    if std::io::stdout().is_terminal() {
+        new_ucmd!().arg("-c").succeeds().code_is(0);
```
Comment (cakebaker) [inline tests/by-util/test_more.rs:16]: `code_is(0)` is not necessary, it's already implied by `succeeds()`.

**Comment by cakebaker** [comment]
While `more -c file.txt` works fine, `more -p file.txt` doesn't seem to work correctly. When I start paging, the content of the previous page(s) is not cleared.

**Comment by Ideflop** [comment]
I have corrected the behavior of the two commands because they were not functioning properly previously. 
The command `-c` mistakenly cleared both the displayed text and its history instead of just deleting the currently displayed text. 
The command `-p` now works correctly by taking the displayed text and adding it to the history. As a result, the terminal is emptied, allowing the remaining content to be printed.

**Comment by cakebaker** [comment]
Thanks for your PR :)

**Pull request merged**